docs/en/ros2/px4_ros2_control_interface.md

```diff
@@ -400,6 +401,135 @@ _goto_setpoint->update(
+const float lateral_acceleration = 2.F; // FRD, used as feedforward
+
+_fw_lateral_longitudinal_setpoint->updateWithAltitude(altitude_msl,
+  course,
+  equivalent_aspd,
+  lateral_acceleration);
+```
+
+The equivalent airspeed and lateral acceleration arguments are defined as `std::optional<float>`, so you can omit any of them by passing `std::nullopt`.
+
+::: tip
+If both lateral acceleration and course setpoints are provided, the lateral acceleration setpoint will be used as feedforward.
+:::
+
+##### Full Control Using the Setpoint Struct
+
+For full flexibility, you can create and pass a `FwLateralLongitudinalSetpoint` struct.
+Each field is templated with `std::optional<float>`.
+
+::: tip
+If both course and airspeed direction are set: airspeed direction takes precedence, course is not controlled.
+Lateral acceleration is treated as feedforward if either course or airspeed direction are also finite.
+If both altitude and height rate are set: height rate takes precedence, altitude is not controlled.
+:::
+
+```cpp
+px4_ros2::FwLateralLongitudinalSetpoint setpoint_s;
+
+setpoint_s.withCourse(0.F);
+// setpoint_s.withAirspeedDirection(0.2F); // uncontrolled
+setpoint_s.withLateralAcceleration(2.F); // feedforward
+//setpoint_s.withAltitude(500.F); // uncontrolled
+setpoint_s.withHeightRate(2.F);
+setpoint_s.withEquivalentAirspeed(15.F);
+
+_fw_lateral_longitudinal_setpoint->update(setpoint_s);
+```
+
+The diagram below illustrates the interaction between the `FwLateralLongitudinalSetpointType` and PX4 when all inputs are set.
+
+![FW ROS Interaction](../../assets/middleware/ros2/px4_ros2_interface_lib/fw_lat_long_ros_interaction.svg)
+
+##### Advanced Configuration (Optional)
+
+You can also pass a `FwControlConfiguration` struct along with the setpoint to override default controller settings and constraints such as pitch limits, throttle limits, and target sink/climb rates.
+This is intended for advanced users:
+
+```cpp
+px4_ros2::FwLateralLongitudinalSetpoint setpoint_s;
+
+setpoint_s.withAirspeedDirection(0.F);
+setpoint_s.withLateralAcceleration(2.F); // feedforward
+setpoint_s.withAltitude(500.F);
+setpoint_s.withEquivalentAirspeed(15.F);
+
+px4_ros2::FwControlConfiguration config_s;
+
+config_s.withTargetClimbRate(3.F);
+config_s.withMaxAcceleration(5.F);
+config_s.withThrottleLimits(0.4F, 0.6F);
+
+_fw_lateral_longitudinal_setpoint->update(setpoint_s, config_s);
+```
+
+All configuration fields are defined as `std::optional<float>`.
+Unset values will default to the PX4 configuration.
+See [LateralControlConfiguration](../msg_docs/LateralControlConfiguration.md) and [FixedWingLongitudinalConfiguration](../msg_docs/LongitudinalControlConfiguration.md) for more information on configuration options.
+
+:::info For safety, PX4 automatically limits configuration values to stay within the vehicle’s constraints. For example, throttle overrides are clamped to remain between [`FW_THR_MIN`](../advanced_config/parameter_reference.md#FW_THR_MIN)
```

Comment:
Indentation matters!!! This would render with the text as a heading.

```suggestion
:::info
For safety, PX4 automatically limits configuration values to stay within the vehicle’s constraints.
For example, throttle overrides are clamped to remain between [`FW_THR_MIN`](../advanced_config/parameter_reference.md#FW_THR_MIN)
```